js/bundle/src/encoder.ts

```diff
@@ -66,6 +69,35 @@ export class BundleBuilder {
     );
   }
 
+  addFile(file: string, url: string) {
+    const headers = {
+      'Content-Type': mime.getType(file) || 'application/octet-stream',
+    };
+    this.addExchange(url, 200, headers, fs.readFileSync(file));
+  }
+
+  addFilesRecursively(dir: string, baseURL: string) {
+    if (!baseURL.endsWith('/')) {
+      throw new Error("baseURL must end with '/'.");
+    }
+    const files = fs.readdirSync(dir);
+    files.sort(); // Sort entries for reproducibility.
+    for (const file of files) {
+      const filePath = path.join(dir, file);
+      if (fs.statSync(filePath).isDirectory()) {
+        this.addFilesRecursively(filePath, baseURL + file + '/');
+      } else if (file === 'index.html') {
+        // If the file name is 'index.html', create an entry for baseURL itself
+        // and another entry for baseURL/index.html which redirects to baseURL.
+        // This matches the behavior of gen-bundle.
+        this.addFile(filePath, baseURL);
+        this.addExchange(baseURL + file, 301, { Location: './' }, '');
```

Comment:
My understanding might be wrong though, if there is a file located at `/foo/bar/baz/index.html`, this code looks redirecting to `baseURL/index.html`. Is that a correct behavior?

Reply:
Note that the recursive call at line 88 updates `baseURL` adding the directory name.

Here, `baseURL`'s value should be `<user-specified-baseURL>/foo/bar/baz/`, so this exchange would redirect `<user-specified-baseURL>/foo/bar/baz/index.html` to `<user-specified-baseURL>/foo/bar/baz/`.

Reply:
Ah! I was missing it! Thank you for the comment!